src/lib/y2storage/dump_manager.rb

```diff
@@ -0,0 +1,327 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "singleton"
+require "fileutils"
+require "yast"
+require "rspec/mocks"
```

Comment:
https://github.com/shundhammer/yast-storage-ng/blob/85bc187c990b1a8cc1c996153e34d2a87814f927/src/lib/y2storage/dump_manager.rb#L101

```Ruby
    def dump
    ....
      elsif dump_obj.is_a?(RSpec::Mocks::InstanceVerifyingDouble) ||
          dump_obj.is_a?(RSpec::Mocks::Double)
        log.warn("Not dumping #{dump_obj.class}")
```
That's the one thing we talked about in today's meeting: This is not exactly pretty, but it helps a lot to keep unit tests simple; otherwise a `double` or `instance_double` for `Y2Storage::Devicegraph` would need the appropriate `save` method every time.